### DISCUSSION TITLE
Motion + Bookworm 64 + Raspberry Pi5

### DISCUSSION THREAD

**Discussion by Tr14rk** [Motion-Project/motion#1753] [Q&A]
Hello,

I'm new to Motion.
My hardware and software:
Raspberry Pi5 Bookworm 64 bit + Camera Module 3
Libcamera installed and working (libcamera-vid records well...).

I installed Bookworm 4.6.0-1 version for arm64: https://github.com/Motion-Project/motion/releases/download/release-4.6.0/bookworm_motion_4.6.0-1_arm64.deb

According to the v4l2-ctl documentation it is necessary to start Motion with:
libcamerify motion
The web server starts fine, but it doesn't show the camera...and it gives me this error:
libcamerify motion
ERROR: ld.so: object '/usr/lib/aarch64-linux-gnu/libcamera/v4l2-compat.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
[0:motion] [NTC] [ALL] conf_load: Processing thread 0 - config file /etc/motion/motion.conf
[0:motion] [NTC] [ALL] motion_startup: Logging to syslog
[0:motion] [NTC] [ALL] motion_startup: Motion 4.6.0 Started
[0:motion] [NTC] [ALL] motion_startup: Using default log type (ALL)
[0:motion] [NTC] [ALL] motion_startup: Using log type (ALL) log level (NTC)
[0:motion] [NTC] [ALL] become_daemon: Motion going to daemon mode

I checked Motion read and write permissions: Ok

I observed this:
1. If I "assign" manually with:
sudo v4l2-ctl --set-fmt-video=width=1920,height=1080,pixelformat=YUYV

2. I run libcamerify motion (nothing is seen on the web server)

3. and I execute v4l2-ctl --all, it always changes the pixel to 'GREY'

The most important thing is that I don't see any stream from camera module 3 on the server!

Here the commands, results and the log:

(1) pi5@pi5:~ $ sudo v4l2-ctl --set-fmt-video=width=1920,height=1080,pixelformat=YUYV
pi5@pi5:~ $ v4l2-ctl --all
Driver Info:
Driver name: rp1-cfe
Card type: rp1-cfe
Bus info: platform:1f00110000.csi
Driver version: 6.1.73
Capabilities: 0xaca00001
Video Capture
Metadata Capture
Metadata Output
Streaming
Extended Pix Format
Device Capabilities
Device Caps: 0x24a00001
Video Capture
Metadata Capture
Streaming
Extended Pix Format
Media Driver Info:
Driver name: rp1-cfe
Model: rp1-cfe
Serial:
Bus info: platform:1f00110000.csi
Media version: 6.1.73
Hardware revision: 0x00114666 (1132134)
Driver version: 6.1.73
Interface Info:
ID: 0x03000017
Type: V4L Video
Entity Info:
ID: 0x00000015 (21)
Name: rp1-cfe-csi2_ch0
Function: V4L2 I/O
Pad 0x01000016 : 0: Sink, Must Connect
Link 0x02000037: from remote pad 0x1000006 of entity 'csi2' (Video Interface Bridge): Data
Priority: 2
Video input: 0 (rp1-cfe-csi2_ch0: ok)
Video Capture Format:
Width/Height: 1920/1080
Pixel Format: 'YUYV' (YUYV 4:2:2)
Field: None
Bytes per Line: 3840
Size Image : 4147200
Colorspace: Default
Transfer Function: Default (maps to Rec. 709)
YCbCr/HSV Encoding: Default (maps to ITU-R 601)
Quantization : Default (maps to Limited Range)
Flags:
Format Metadata Capture:
Sample Format: 'SENS' (Sensor Ancillary Metadata)
Buffer Size: 8192

(2) pi5@pi5:~ $ sudo libcamerify motion
ERROR: ld.so: object '/usr/lib/aarch64-linux-gnu/libcamera/v4l2-compat.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
[0:motion] [NTC] [ALL] conf_load: Processing thread 0 - config file /etc/motion/motion.conf
[0:motion] [NTC] [ALL] motion_startup: Logging to syslog
[0:motion] [NTC] [ALL] motion_startup: Motion 4.6.0 Started
[0:motion] [NTC] [ALL] motion_startup: Using default log type (ALL)
[0:motion] [NTC] [ALL] motion_startup: Using log type (ALL) log level (NTC)
[0:motion] [NTC] [ALL] become_daemon: Motion going to daemon mode

(3) pi5@pi5:~ $ v4l2-ctl --all
Driver Info:
	Driver name      : rp1-cfe
	Card type        : rp1-cfe
	Bus info         : platform:1f00110000.csi
	Driver version   : 6.1.73
	Capabilities     : 0xaca00001
		Video Capture
		Metadata Capture
		Metadata Output
		Streaming
		Extended Pix Format
		Device Capabilities
	Device Caps      : 0x24a00001
		Video Capture
		Metadata Capture
		Streaming
		Extended Pix Format
Media Driver Info:
	Driver name      : rp1-cfe
	Model            : rp1-cfe
	Serial           : 
	Bus info         : platform:1f00110000.csi
	Media version    : 6.1.73
	Hardware revision: 0x00114666 (1132134)
	Driver version   : 6.1.73
Interface Info:
	ID               : 0x03000017
	Type             : V4L Video
Entity Info:
	ID               : 0x00000015 (21)
	Name             : rp1-cfe-csi2_ch0
	Function         : V4L2 I/O
	Pad 0x01000016   : 0: Sink, Must Connect
	  Link 0x02000037: from remote pad 0x1000006 of entity 'csi2' (Video Interface Bridge): Data
Priority: 2
Video input : 0 (rp1-cfe-csi2_ch0: ok)
Format Video Capture:
	Width/Height      : 1920/1080
	Pixel Format      : 'GREY' (8-bit Greyscale)
	Field             : None
	Bytes per Line    : 1920
	Size Image        : 2073600
	Colorspace        : Default
	Transfer Function : Default (maps to Rec. 709)
	YCbCr/HSV Encoding: Default (maps to ITU-R 601)
	Quantization      : Default (maps to Full Range)
	Flags             : 
Format Metadata Capture:
	Sample Format   : 'SENS' (Sensor Ancillary Metadata)
	Buffer Size     : 8192

	Another comment is that v4l2-compat.so is not present on the device, despite having reinstalled libcamera-utils:
is this file necessary? It is normal?

Does anyone have this problem?

How could they solve it?

I need help please!

**Comment by untrium** [comment]
Yes, I also have this problem.  I have never succeeded in getting anything more than a grey rectangle from my Camera 3.  I am using a Pi3 but the OS is a fresh install and I am using the latest download from the Motion Project repo.  If I find a solution I'll post it here.

**Comment by Mr-Dave** [comment]
The latest versions have  been tested to work with both libcamerify (Motion 4.7 and prior) as well as with libcamera(Motion 5.0).  The message 

> ERROR: ld.so: object '/usr/lib/aarch64-linux-gnu/libcamera/v4l2-compat.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.

means that the required libraries for libcamera have not been installed.  The package that includes libcamerify has changed over the years so you can try to install the following and see if it resolves the issue.

`sudo apt install libcamera-tools libcamera-dev libcamera-v4l2`

**Reply by DifferentComputers** [reply]
> The latest versions have been tested to work with both libcamerify (Motion 4.7 and prior) as well as with libcamera(Motion 5.0). The message
> 
> > ERROR: ld.so: object '/usr/lib/aarch64-linux-gnu/libcamera/v4l2-compat.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
> 
> means that the required libraries for libcamera have not been installed. The package that includes libcamerify has changed over the years so you can try to install the following and see if it resolves the issue.
> 
> `sudo apt install libcamera-tools libcamera-dev libcamera-v4l2`

Sure would be great if there was a step by step install guide. If there is one, I can't find it. And I have exactly the problem the other posters in this thread have. Libcamera sees the camera. Motion doesn't.
